server/api/batch_inventory.py

```diff
@@ -378,14 +384,82 @@ def validate_choice_name_and_get_choice_id(choice_name: str) -> Optional[str]:
                     )
 
                     batch_key: BatchKey = ("", batch)
-                    choice_id = validate_choice_name_and_get_choice_id(choice_name)
+                    choice_id = validate_choice_name_and_get_choice_id(
+                        contest, choice_name
+                    )
 
                     ballot_count_by_batch[batch_key] += 1
```

Comment:
I know this is existing code, but is it supposed to count a ballot once for each contest (seems like double counting potentially)? I noticed the Hart code you wrote doesn't do this - it counts before looping over contests.

Reply:
Maybe we need some tests for the ES&S code. I know it's not really in scope but I'm worried about it now.

Reply:
yeah the es&s code seemed pretty questionable to me lol, don't want to put adding a test for that in scope here but probably a good followup

Reply:
Sounds good, added a high pri task: https://github.com/votingworks/arlo/issues/2037

Reply:
I claim full responsibility for the questionable code written in a rush 🙈 I should've surfaced earlier that this was worth revisiting and long had on my to-do to write tests for this but ultimately never found the time.

I think you're totally right that this is a bug, one that was masked by the fact that we only expected it to be used for a single-contest audit and manually tested as such. One more relevant piece of context is that it the batch audit file prep tool ultimately wasn't used in the audit that we built ES&S support for. So this code hasn't actually been used.